frontend/components/GameVisibilityList.tsx

```diff
@@ -0,0 +1,46 @@
+import * as React from 'react';
+import { connect } from 'react-redux';
+// eslint-disable-next-line no-unused-vars
+import {AppStore} from '../storage/common';
+interface GameVisibilityListProps{
+	gameVisibilityToggleMap: Map<number, GameVisibilityToggleProps>;
+}
+const GamesVisibilityList: React.FC<GameVisibilityListProps> = (props: GameVisibilityListProps) =>{
+	const {gameVisibilityToggleMap: gamesVisibility} = props;
+
+	return (
+		<>
+			{Array.from(gamesVisibility.values()).map(game=> {
+				return <GameVisibilityToggle key={game.id} name={game.name} isVisible={game.isVisible} id={game.id} />;
+			})}
+		</>
+	);
+};
+interface GameVisibilityToggleProps {
+	id: number;
+	name: string;
+	isVisible: boolean;
+}
+
+const GameVisibilityToggle: React.FC<GameVisibilityToggleProps> = (props) => <>
+	<div>
+		<input type="checkbox" checked = {props.isVisible} name={props.name} id={`spoiler #${props.id}`}/> {props.name}
+	</div>
+
+</>;
+
+const mapStateToProps = (state: AppStore): GameVisibilityListProps => {
+	const games = state.games;
+	const gamesVisibility = state.gamesVisibility;
+	const gameVisibilityToggleMap = new Map<number, GameVisibilityToggleProps>();
+	Array.from(games.values()).map(game =>{
```

Comment:
`Array.map()` is used to [transform an array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map). What you're doing here is just iterating over a map, so `forEach` would work better.

But you don't need to do that either, because you can just as well use `for (const game of games.values())` ([for...of docs](https://www.typescriptlang.org/docs/handbook/iterators-and-generators.html)) and avoid the awkward *casting* to array.